test/Recollateralization.test.ts

```diff
@@ -3088,6 +3113,141 @@ describe(`Recollateralization - P${IMPLEMENTATION}`, () => {
         expect(await rsr.balanceOf(stRSR.address)).to.equal(stakeAmount.sub(sellAmtRemToken))
         expect(await stRSR.balanceOf(addr1.address)).to.equal(stakeAmount)
       })
+
+      context('DutchTrade', () => {
+        const auctionLength = 300
+        beforeEach(async () => {
+          await broker.connect(owner).setDutchAuctionLength(auctionLength)
+
+          // Set up rebalancing scenario
+          await basketHandler.connect(owner).setPrimeBasket([token1.address], [fp('1')])
+          await basketHandler.refreshBasket()
+        })
+
+        it('Should not trade when paused', async () => {
+          await main.connect(owner).pauseTrading()
+          await expect(backingManager.rebalance(TradeKind.DUTCH_AUCTION)).to.be.revertedWith(
+            'frozen or trading paused'
+          )
+        })
+
+        it('Should not trade when frozen', async () => {
+          await main.connect(owner).freezeLong()
+          await expect(backingManager.rebalance(TradeKind.DUTCH_AUCTION)).to.be.revertedWith(
+            'frozen or trading paused'
+          )
+        })
+
+        it('Should trade if issuance paused', async () => {
+          await main.connect(owner).pauseIssuance()
+          await expect(backingManager.rebalance(TradeKind.DUTCH_AUCTION)).to.emit(
+            backingManager,
+            'TradeStarted'
+          )
+        })
+
+        it('Should only run 1 trade at a time', async () => {
+          await backingManager.rebalance(TradeKind.DUTCH_AUCTION)
+          await expect(backingManager.rebalance(TradeKind.DUTCH_AUCTION)).to.be.revertedWith(
+            'already rebalancing'
```

Comment:
we should add another scenario that explicitly checks for `already rebalancing` in the "buffer" block

Reply:
ie. run dutch auction => jump to tradeEnd[DUTCH_AUCTION] + ONE_BLOCK => try to rerun dutch auction and check reverted with "already rebalancing"

Reply:
it's good you suggested this because the require condition was actually backwards, adding ONE_BLOCK to the wrong side